src/components/breakpointHandler.ts

```diff
@@ -0,0 +1,667 @@
+
+            let range;
+            for (const node of nodes) {
+                range = getCppRange(
+                    node,
+                    cachePath + "/map/map_cpp.json"
+                )
+                if (range)
+                    break;
+            }
+
+            if (!range || !range.from)
+                continue;
+
+            return {
+                line: range.from,
+                cachePath: cachePath,
+                functionName: currentFunc.name,
+                target: currentFunc.target_name
+            }
+        }
+
+        return undefined;
+    }
+
+    public handleNodeAdded(node: Node, sdfgName: string) {
+        // Creating a Breakpoint for the BP panel
+        /* let uri = vscode.Uri.parse(
+            'C:/Users/Benjamin/Documents/Code/Bachelor/dace/.dacecache/myprogram/program.sdfg'
+        );
+        let pos = new vscode.Position(5, 0);
+        let location = new vscode.Location(uri, pos);
+        let new_bp = new vscode.SourceBreakpoint(location);
+        vscode.debug.addBreakpoints([new_bp]); */
+
+        // Search for the file with the corresponding function information
+        Object.values(this.files).forEach(functions => {
+            let funcDetails = functions.find(func => {
+                return func.name === sdfgName;
+            });
+
+            if (funcDetails) {
+                node.cache = funcDetails.cache;
+                node.sdfg_name = funcDetails.name;
+                node.target = funcDetails.target_name;
+
+                let range = getCppRange(
+                    node,
+                    node.cache + "/map/map_cpp.json"
+                );
+
+                if (!range || !range.from) {
+                    DaCeVSCode.getInstance().getActiveEditor()?.postMessage({
+                        'type': 'unbound_breakpoint',
+                        'node': node
+                    });
+                    return;
+                }
+
+                if (!this.savedNodes[sdfgName])
+                    this.savedNodes[sdfgName] = [];
+                this.savedNodes[sdfgName].push(node);
+                return;
+            }
+        });
+    }
+
+    public handleNodeRemoved(node: Node, sdfgName: string) {
+        if (this.savedNodes[sdfgName])
+            this.savedNodes[sdfgName].forEach((n, i, _) => {
+                if (node.isEqual(n)) {
+                    this.savedNodes[sdfgName].splice(i, 1);
+                    return;
+                }
+            });
+    }
+
+    private getNode(line: number, path: fs.PathLike): Node[] | undefined {
+
+        let mapPy = jsonFromPath(path);
+        if (!mapPy) {
+            return undefined;
+        }
+
+        let nodeJSON = mapPy[line.toString()];
+        // Return undefined if the line doesn't exist in the mapping
+        if (!nodeJSON) {
+            return undefined;
+        }
+
+        // Make sure the JSON object has the right Properties
+        /* if (
```

Comment:
Commented out intentionally?

Reply:
I just replaced it with a try and catch in case the source map has the wrong format